src/Swarm/TUI/View.hs

```diff
@@ -142,15 +144,71 @@ drawTPS s = hBox (tpsInfo : rateInfo)
 replHeight :: Int
 replHeight = 10
 
+-- | Hide the cursor when a modal is set
+chooseCursor :: AppState -> [CursorLocation n] -> Maybe (CursorLocation n)
+chooseCursor s locs = case s ^. uiState . uiModal of
+  Nothing -> showFirstCursor s locs
+  Just _ -> Nothing
+
 -- | The error dialog window.
 errorDialog :: Dialog ()
 errorDialog = dialog (Just "Error") Nothing 80
 
+-- | Render a fullscreen widget with some padding
+renderModal :: Modal -> Widget Name
+renderModal modal = renderDialog (dialog (Just modalTitle) Nothing 500) modalWidget
+  where
+    modalWidget = Widget Fixed Fixed $ do
+      ctx <- getContext
+      let w = ctx ^. availWidthL
+          h = ctx ^. availHeightL
+          padding = 10
+      render $ setAvailableSize (w - padding, h - padding) modalContent
+    (modalTitle, modalContent) =
+      case modal of
+        HelpModal -> ("Help", helpWidget)
+
+helpWidget :: Widget Name
+helpWidget = (helpKeys <=> fill ' ') <+> (helpCommands <=> fill ' ')
+  where
+    helpKeys =
+      vBox [ hCenter $ txt "Global keybindings"
+           , hCenter $ mkTable glKeyBindings
+           ]
+    mkTable = BT.renderTable . BT.table . map toWidgets
+    toWidgets (k, v) = [txt k, txt v]
+    glKeyBindings =
+      [ ("F1", "Help")
+      , ("Ctrl-q", "quit the game")
+      , ("Tab", "cycle panel focus")
+      , ("Meta-w", "focus on the world map")
+      , ("Meta-e", "focus on the info")
+      , ("Meta-r", "focus on the REPL")
+      ]
+    helpCommands =
+      vBox [ hCenter $ txt "Base commands"
+           , hCenter $ mkTable baseCommands
+           , padTop (Pad 1) $ hCenter $ txt "Robot commands"
+           , hCenter $ mkTable robotCommands
+           ]
+    baseCommands =
+      [ ("build <name> <commands>", "Create a robot")
```

Comment:
I'm not sure we want to make a distinction in the help between "base commands" and "robot commands".  There is actually no distinction as far as the game is concerned, other than the fact that they start out with different sets of devices.  But, for example, it is already pretty easy to craft another workbench which you can install in a robot so that it can use the `make` command.  And the base itself can execute the `grab` command (actually if you try this, you realize that it is sitting on top of a tree =).  In the future, we can imagine having robots `build`ing more robots.  I also think it would be fun if eventually you can do something like install treads on your base so it can `move`.

Reply:
Alright, that also simplifies the layout :)

Reply:
Great. =)  And maybe just call it "commands" instead of "base commands".

Reply:
Done.